File: Java/Aplicatii client-server/labJavaClientServerServiciiRest/Services/src/main/java/services/IServices.java
package services;

import model.Cursa;
import model.Oficiu;
import model.Participant;
import model.ParticipantCursa;

public interface IServices {

    Oficiu findOficiuByUserPass(String username, String password, IObserver client) throws MyException;
    void logout(Oficiu oficiu, IObserver client) throws MyException;
    Cursa[] findAllCursa() throws MyException;
    Cursa findCursa(String idCursa) throws MyException;
    void updateCursa(String idCursa, Cursa cursa) throws MyException;
    Participant[] findAllParticipant(String echipa) throws MyException;
    Participant[] findAllParticipant() throws MyException;
    Participant findParticipant(String nume, String echipa, Integer capMotor) throws MyException;
    String findAllEchipe() throws MyException;
    void saveParticipantCursa(ParticipantCursa participantCursa) throws MyException;

    Oficiu findOficiuByUserPass(String username, String password) throws MyException;

    void setClient(IObserver client);
}

File: C#/Aplicatii client-server/labClientServerCSharpProtoBuf/ClientServer/Services/IServices.cs
using lab3.domain;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Services
{
    public interface IServices
    {
        Oficiu findOficiuByUserPass(string username, string password, IObserver client);
        Oficiu findOficiuByUserPass(string username, string password);
        void login(Oficiu oficiu, IObserver client);
        void logout(Oficiu oficiu, IObserver client);
        Cursa findCursa(string id);
        Cursa[] findAllCursa();
        Cursa[] findAllCursa(int capMotor);
        Participant[] findAllParticipant(string echipa);
        Participant[] findAllParticipant();
        Participant findParticipant(string nume, string echipa, int capMotor);
        void saveParticipantCursa(ParticipantCursa participantCursa);
        void saveCursa(Cursa cursa);
        void deleteCursa(string idCursa);
        void submitted(Cursa[] cursa);
    }
}
